File: tkpdinbox/src/main/java/com/tokopedia/inbox/common/util/SpaceItemDecoration.java
package com.tokopedia.inbox.common.util;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * @author by nisie on 3/12/18.
 */

public class SpaceItemDecoration extends RecyclerView.ItemDecoration {

    private int space;

    public SpaceItemDecoration(int verticalSpaceHeight) {
        this.space = verticalSpaceHeight;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
                               RecyclerView.State state) {

        outRect.right = space / 2;
        outRect.left = space / 2;
    }
}

File: features/category/catalog_common/src/main/java/com/tokopedia/catalogcommon/util/decoration/SpaceItemDecoration.kt
package com.tokopedia.catalogcommon.util.decoration

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class SpaceItemDecoration(private val space: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        val position = parent.getChildAdapterPosition(view)

        outRect.top = space
        outRect.bottom = space

        if (position == 0) {
            outRect.left = 0
            outRect.right = 0
        } else if (position == parent.adapter?.itemCount?.minus(1)) {
            outRect.left = 0
            outRect.right = 0
        } else {
            outRect.left = space
            outRect.right = space
        }
    }
}
